asciidoc/courses/llm-fundamentals/modules/2-vectors-semantic-search/lessons/4-improving-semantic-search/lesson.adoc

```diff
@@ -0,0 +1,83 @@
+
+The context from which the similarity scores are provided is an important factor here.
+Our embeddings are trained on a generic dataset, and as such, may identify two movies with a main character called Jack who likes fruit as movies with a close similarity.
+In one sense, this is correct, as these movies will be similar concepts when compared to, say, a vanilla yoghurt or a cricket bat.
+
+On the other hand, if you are recommending a childrens movie to pass away a rainy afternoon, *Jack and the Beanstalk* is worlds away from *The Shining*.
+
+
+== Incorporating Graph Features
+
+Graph features can enrich vector-backed semantic search by providing structural and relational context to data.
+Leveraging the interconnectedness inherent in graphs, semantic search can draw upon relationships and hierarchies between entities, enhancing the depth and relevance of search results.
+While vectors capture semantic nuances, the addition of graph structures allows for a more holistic understanding of data, considering both its inherent meaning and its position in a broader network of knowledge.
+
+This topic is out of the scope of this module, but for more information, we recommend watching link:https://www.youtube.com/watch?v=bRD09ndyJNs[Going Meta - Ep 21: Vector-based Semantic Search and Graph-based Semantic Search^], in which Dr Jesus Barassa and Alexander Erdl explore the differences between Vector-based Semantic Search and Graph-based Semantic Search.
+
+== Using Feedback
+
+A simpler approach may be to use User feedback to gradually re-rank and curate the content returned by a vector index search.
+This is the approach we take with the GraphAcademy chatbot.
+
+To celebrate the launch of the Vector Index, we published an article detailing link:https://medium.com/neo4j/building-an-educational-chatbot-for-graphacademy-with-neo4j-f707c4ce311b[how we used vector index to identify Neo4j content] to improve LLM responses.
+The article demonstrates how we ingested Neo4j Documentation and GraphAcademy lessons, using embeddings of the content to populate a vector index.
+
+// TODO: Move to CDN?
+image::https://miro.medium.com/v2/resize:fit:4800/format:webp/1*BpDS376rEKb6kWBlxNobRQ.png[The GraphAcademy Chatbot Data Model]
+
+When a user asks a question, the server generates an embedding of the user's question and uses the vector index to identify relevant content.
+
+This approach works well in most cases, but now and again, the index fails to return relevant content, in which case the LLM is instructed to respond to the user with an apology.
+
+When this happens, or the user provides feedback, this is stored in the database.
+This means over time, we get a better picture of which documents do or do not answer specific queries.
+
+When a new user question comes in, we can compare this question to previous questions, and exclude any suggested documents where the response was considered unhelpful.
+
+[source,cypher,rel=noplay]
```

Comment:
jump in complexity is pretty high, perhaps we should start with the simple example of filtering for the category of family movies genre in the first example